### PR TITLE
Fix encrypted live stream handling

### PR DESCRIPTION
**Pull request by RenierM26**
## Summary
- match EZVIZ native transform behavior by decrypting only the encrypted 4096-byte prefix of each NAL body
- send proactive VTM stream keepalives while media packets are flowing, instead of only responding to server keepalive requests
- add regression coverage for both behaviors

## Tests
- `.venv/bin/pytest tests/test_stream.py`
- `.venv/bin/pytest`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

### CODE REVIEW

File: pyezvizapi/stream.py [hunk 1]
```diff
@@ -1080,24 +1095,32 @@ def decrypt_hikvision_ps_video(
...
+        active_nal_decrypted = 0
 
     def decrypt_nal_body_segment(start: int, end: int) -> None:
+        nonlocal active_nal_decrypted
         if end <= start:
             return
-        for pos in range(start, end):
+        remaining = HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH - active_nal_decrypted
+        if remaining <= 0:
+            return
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1253]: Fixed in fcb1dfd. The decryptor now stops considering later Annex-B-looking markers in a continued NAL once the native 4096-byte encrypted prefix has been consumed, unless a fresh PES payload begins with a start code. Added a regression where the preserved ciphertext tail contains a plausible `00 00 01 42 01` lookalike.

File: pyezvizapi/stream.py [hunk 2]
```diff
@@ -1120,6 +1143,12 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
             continue
 
         for idx, (start_code_pos, start_code_len) in enumerate(nal_starts):
+            if (
+                active_nal
+                and active_nal_decrypted >= HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos != payload_start
+            ):
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1341]: Fixed in 9525a35. The guard now skips short `00 00 01` lookalikes after the 4096-byte encrypted prefix instead of breaking out of the NAL scan, so a following real long Annex-B NAL in the same PES still resets state and decrypts its own prefix. Added regression coverage for that legitimate subsequent-NAL case.

File: pyezvizapi/stream.py [hunk 3]
```diff
@@ -1120,6 +1143,13 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
             continue
 
         for idx, (start_code_pos, start_code_len) in enumerate(nal_starts):
+            if (
+                active_nal
+                and active_nal_decrypted >= HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_len == len(MPEG_START_CODE_PREFIX)
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 932edb9. The NAL state now tracks the body start offset and skips an Annex-B marker that appears exactly where the native 4096-byte encrypted prefix ends, regardless of whether it uses the three-byte or four-byte prefix form. That keeps `00 00 00 01` tail lookalikes preserved as continued encrypted tail data while the existing regression for a later real NAL in the same PES still passes.

Validation:
- `.venv/bin/pytest tests/test_stream.py`
- `.venv/bin/pytest`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 4]
```diff
@@ -1120,11 +1143,20 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
             continue
 
         for idx, (start_code_pos, start_code_len) in enumerate(nal_starts):
+            if (
+                active_nal
+                and active_nal_decrypted >= HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos
+                == active_nal_body_start + HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 0a2cee4. The decryptor now treats later Annex-B candidates in a long NAL's preserved tail as ciphertext lookalikes once the native 4096-byte decrypt prefix has been consumed, while still allowing a following NAL with enough body to carry its own encrypted prefix. Added regression coverage for a later `00 00 00 01` tail lookalike.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

**Comment by RenierM26** [comment]
Addressed the latest Codex review in eb13447 (`Allow short NALs after long NALs`). The broad candidate-body-length heuristic from 0a2cee4 was removed because it could suppress legitimate short NALs after a long NAL. The scanner now only skips the concrete boundary lookalike case, and the real-following-NAL regression is back to the short encrypted body case.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 5]
```diff
@@ -1120,11 +1143,20 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
+                and start_code_pos != payload_start
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 2061789. The scanner now preserves a start code at exactly `active_nal_body_start + HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH` as a real NAL boundary, and only treats later candidates as preserved-tail lookalikes once actual tail bytes have begun. Added a regression for the exact-prefix boundary case.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 6]
```diff
@@ -1067,37 +1081,48 @@ def decrypt_hikvision_ps_video(
...
     output = bytearray(data)
     pending_block_positions: list[int] = []
     pending_block = bytearray()
     active_nal = False
+    active_nal_decrypted = active_nal_body_start = 0
     find_nal_start_codes = (
-        _find_h264_nal_start_codes
+        _find_nal_start_codes
+        if nalu_header_size == 0
+        else _find_h264_nal_start_codes
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1235]: Fixed in 2061789. For encrypted-header H.264 mode (`nalu_header_size == 0`), candidates found before the active NAL has consumed the 4096-byte encrypted prefix are kept as encrypted payload bytes instead of being treated as fresh Annex-B boundaries. Added a regression where the encrypted first block begins with `00 00 01` and still decrypts with the correct AES alignment.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 7]
```diff
@@ -1120,18 +1145,37 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
+                and start_code_pos != payload_start
+            ):
+                continue
+            if (
+                active_nal
+                and active_nal_decrypted >= HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos
+                > active_nal_body_start + HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos != payload_start
+            ):
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1341]: Fixed in 8a9da07. The tail-lookalike guard no longer exempts candidates just because they appear at the start of a later PES payload; candidates after `active_nal_body_start + HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH` are preserved as continued tail data even across PES header gaps. Added a regression where the second video PES starts with a plausible `00 00 00 01 42 01` tail lookalike and remains untouched.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 8]
```diff
@@ -1120,18 +1145,36 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
 
         for idx, (start_code_pos, start_code_len) in enumerate(nal_starts):
+            if (
+                active_nal
+                and nalu_header_size == 0
+                and active_nal_decrypted + max(0, start_code_pos - segment_start)
+                < HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos != payload_start
+            ):
+                continue
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in b932c8e. In encrypted-header H.264 mode, in-prefix candidates are no longer skipped unconditionally: the scanner now keeps suppressing zero-byte/misaligned ciphertext lookalikes, but accepts a candidate that lands on an AES block boundary and whose first decrypted byte is a plausible H.264 NAL header. Added a regression with two 16-byte encrypted H.264 NALs in one PES so the second NAL resets AES alignment and decrypts correctly.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 9]
```diff
@@ -1120,18 +1153,47 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
             )
+            if (
+                active_nal
+                and nalu_header_size == 0
+                and (
+                    candidate_decrypted := active_nal_decrypted
+                    + max(0, start_code_pos - segment_start)
+                )
+                < HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos != payload_start
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 0e104c8. The encrypted-header H.264 pre-prefix lookalike filter now also applies at PES payload starts, so a continued NAL split across adjacent video PES packets keeps decrypting with the existing AES alignment instead of resetting on ciphertext bytes that look like `00 00 01`. Added regression coverage where the second PES starts with that exact encrypted continuation pattern.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 10]
```diff
@@ -1120,18 +1153,46 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
+                active_nal
+                and nalu_header_size == 0
+                and (
+                    candidate_decrypted := active_nal_decrypted
+                    + max(0, start_code_pos - segment_start)
+                )
+                < HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and (
+                    candidate_decrypted == 0
+                    or candidate_decrypted % AES.block_size != 0
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 4d999ce. The encrypted-header H.264 pre-prefix guard no longer requires the previous NAL body length to end on an AES block boundary. It still suppresses zero-byte candidates and candidates whose first encrypted block does not decrypt to a plausible H.264 NAL header, but accepts real Annex-B boundaries after unaligned NAL bodies. Added a regression where the first H.264 NAL leaves a partial encrypted tail unchanged and the following NAL resets/decrypts correctly.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `git diff --check`

File: pyezvizapi/__main__.py [hunk 11]
```diff
@@ -1514,18 +1541,26 @@ def __init__(self, key: str, *, codec: str) -> None:
         self._nalu_header_size = _codec_nalu_header_size(codec)
         self._buffer = bytearray()
 
+    def _decrypt_chunk(self, chunk: bytes) -> bytes:
+        if self._nalu_header_size is None:
+            self._nalu_header_size = detect_hikvision_ps_video_nalu_header_size(
+                chunk,
+                self._key,
+            )
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/__main__.py:1578]: Fixed in 8eb3729. `detect_hikvision_ps_video_nalu_header_size()` now accepts `default=None` so callers can distinguish “no video NAL evidence yet” from an actual HEVC decision. The buffered stream decryptor uses that deferred mode: initial non-video PS prefixes are processed without locking auto mode, and the codec is only stored after a chunk with detectable video NAL evidence. Added regression coverage for both the detector no-evidence path and the buffered decryptor deferral/lock behavior.

Validation:
- `.venv/bin/pytest tests/test_stream.py tests/test_cli.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py pyezvizapi/__main__.py`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 12]
```diff
@@ -1120,18 +1251,45 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
+                    )
+                )
+            ):
+                continue
+            if (
+                active_nal
+                and active_nal_decrypted >= HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+                and start_code_pos
+                > active_nal_body_start + HIKVISION_NAL_ENCRYPTED_PREFIX_LENGTH
+            ):
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1341]: Fixed in bebbf08. `decrypt_hikvision_ps_video()` now iterates complete MPEG-PS packet ranges instead of only video payload ranges, and resets the active NAL/AES state when it crosses an audio/private/non-video packet. Adjacent video PES packets still carry NAL state, so split NALs continue to decrypt correctly, but interleaved PS packets now break the NAL run before the next real video NAL.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 13]
```diff
@@ -1111,7 +1237,23 @@ def decrypt_nal_body_segment(start: int, end: int) -> None:
...
-    for payload_start, payload_end in _video_pes_payload_ranges(data):
+    def starts_plausible_encrypted_h264_nal(start: int, end: int) -> bool:
+        if end - start < AES.block_size:
+            return False
+        cipher = AES.new(aes_key, AES.MODE_CBC, iv=bytes(AES.block_size))
+        decrypted_header = cipher.decrypt(bytes(output[start : start + AES.block_size]))[0]
+        nal_type = decrypted_header & 0x1F
+        return 1 <= nal_type <= 23
+
+    for packet_range in _mpeg_ps_complete_packet_ranges(data):
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 6b9f07f. `_mpeg_ps_complete_packet_ranges()` remains conservative by default for incremental buffering, but `decrypt_hikvision_ps_video()` now opts in to treating a trailing zero-length video PES as ending at `len(data)`. That lets cloud decrypt and live `flush()`/duration-limited captures decrypt the final unbounded video payload without prematurely flushing open video runs during streaming.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 14]
```diff
@@ -1025,6 +1052,100 @@ def _h264_nal_type(data: bytes, start_code_pos: int, start_code_len: int) -> int
...
+        for start_code_pos, start_code_len in _find_nal_start_codes(
+            data,
+            payload_start,
+            payload_end,
+        ):
+            header_pos = start_code_pos + start_code_len
+            if _is_plausible_hevc_header(data, start_code_pos, start_code_len):
+                scores["hevc"] += _hevc_header_score(
+                    (data[header_pos] >> 1) & 0x3F,
+                )
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py:1139]: Fixed in a3bbf85. HEVC auto-detection now requires the normal single-layer `nuh_layer_id == 0`, so a clear-header H.264 P-slice byte like `0x41` is no longer scored as a HEVC VPS. The detector also no longer runs the encrypted-header H.264 probe for a candidate that already has a plausible clear H.264 header, which prevents the same clear P-slice from being outweighed by a false encrypted-header score.

Added regression coverage for a clear-header H.264 P-slice detection chunk and verified the auto decrypt path keeps `nalu_header_size=1`.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

File: pyezvizapi/stream.py [hunk 15]
```diff
@@ -1025,6 +1052,106 @@ def _h264_nal_type(data: bytes, start_code_pos: int, start_code_len: int) -> int
...
+            has_clear_h264_header = _is_plausible_h264_header(
+                data,
+                start_code_pos,
+                start_code_len,
+            )
+            if has_clear_h264_header:
+                scores["h264-clear-header"] += _h264_header_score(
+                    data[header_pos] & 0x1F,
+                )
+            if not has_clear_h264_header and header_pos + AES.block_size <= payload_end:
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/stream.py]: Fixed in 3ef5365. Auto detection now always runs the encrypted-header H.264 probe when a full AES block is available, even if the visible byte is syntactically valid as a clear H.264 header. To keep the earlier clear P-slice fix intact, common clear H.264 NAL types now score strongly enough that an ambiguous encrypted-header probe tie does not override them.

Added regression coverage for a ciphertext first byte `0x44` that previously looked like a clear H.264 NAL header but decrypts as encrypted-header H.264.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

**Comment by RenierM26** [comment]
Addressed the two latest Codex review threads in 9048d55 (`Suppress in-prefix NAL lookalikes`).

- The in-prefix lookalike guard now applies to clear-header modes too, so HEVC and h264-clear candidates found before the active NAL has consumed its native 4096-byte encrypted prefix are treated as encrypted payload continuation instead of fresh NAL boundaries.
- Auto detection now lets encrypted-header H.264 win ties against HEVC while still requiring it to beat the clear-H.264 score, so clear H.264 P-slices remain classified as clear-header H.264.

Added regressions for both cases.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

**Comment by RenierM26** [comment]
Addressed the latest Codex review in 0c718f4 (`Preserve short clear-header NAL boundaries`).

The clear-header in-prefix suppression now only treats very early candidates, before a full AES block has been consumed, as ciphertext lookalikes. That preserves real short HEVC/h264-clear NAL boundaries after a short encrypted body while keeping the previous in-prefix ciphertext lookalike regression covered.

Added regression coverage for two short HEVC NALs in one PES where the first encrypted body is below the 4096-byte native prefix limit.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

**Comment by RenierM26** [comment]
Addressed the latest Codex review in 5a8bba6 (`Preserve sub-block clear-header NAL boundaries`).

For HEVC/h264-clear modes, the decryptor now trusts codec-specific Annex-B boundaries inside the native 4096-byte prefix once an active NAL has made any progress, including sub-block bodies. That prevents pending partial AES bytes from being carried across real NAL boundaries. The encrypted-header H.264 lookalike guard remains stricter because that mode lacks clear codec headers.

I also adjusted the regression coverage to include a sub-block HEVC NAL followed by another NAL in the same PES.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

File: pyezvizapi/__main__.py [hunk 16]
```diff
@@ -1500,10 +1521,16 @@ def _decrypt_stream_payload_bytes(
...
     """Return the Annex B NAL header length preserved by the decrypt transform."""
 
-    return 2 if codec == "hevc" else 1
+    if codec == "auto":
+        return None
+    if codec == "hevc":
+        return 2
+    if codec == "h264-clear-header":
+        return 1
+    return 0
```
[the remark this answers is not in the document]
Comment (RenierM26) [reply pyezvizapi/__main__.py]: Fixed in be5f0d1 (`Keep h264 decrypt codec compatibility`).

`--decrypt-codec h264` now maps back to the original clear-header H.264 behavior (`nalu_header_size=1`), and the encrypted-header mode is exposed separately as `--decrypt-codec h264-encrypted-header`. I updated the CLI choices/help, README, changelog, and regression tests for both paths.

Validation:
- `.venv/bin/pytest tests/test_cli.py::test_cloud_video_decrypt_can_use_explicit_key_without_login tests/test_cli.py::test_cloud_video_decrypt_h264_keeps_clear_header_mapping -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/__main__.py`
- `git diff --check`

**Comment by RenierM26** [comment]
Addressed the latest Codex review in 7bb88ff (`Preserve decrypt state across PS metadata`).

The decryptor now treats MPEG-PS pack/system/map/padding packets as container metadata instead of elementary-stream boundaries. Those packets no longer reset active NAL/AES state, and `mpeg_ps_decryptable_prefix_length()` also keeps them inside a trailing video PES run so buffered live decrypt does not flush a split NAL early. Audio/private/non-video PES packets still reset state.

Added regressions for a split NAL with a pack header between video PES packets, plus the buffered-prefix case where metadata appears inside the trailing video run.

Validation:
- `.venv/bin/pytest tests/test_stream.py::test_decrypt_hikvision_ps_video_carries_nal_body_across_pack_header tests/test_stream.py::test_decrypt_hikvision_ps_video_resets_after_non_video_packets tests/test_stream.py::test_mpeg_ps_decryptable_prefix_keeps_metadata_in_trailing_video_run -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

The inline reply failed after the push because GitHub could not find the parent comment, but the review thread is resolved.

**Comment by RenierM26** [comment]
Addressed the latest Codex review in bd68662 (`Keep HEVC auto-detect on probe ties`).

Encrypted-header H.264 now has to beat the HEVC score instead of tying it before auto-detection returns `nalu_header_size=0`. That keeps real HEVC VPS/SPS/PPS evidence from being overridden by a random misaligned encrypted-H.264 probe, while still allowing encrypted-header H.264 to win when its probe score is stronger than HEVC and clear-header H.264 evidence.

I replaced the previous tie regression with a HEVC sample whose misaligned encrypted-H.264 probe scores as a common H.264 type; auto mode now keeps `nalu_header_size=2` and decrypts the HEVC body correctly.

Validation:
- `.venv/bin/pytest tests/test_stream.py::test_detect_hikvision_ps_video_nalu_header_size_identifies_hevc tests/test_stream.py::test_detect_hikvision_ps_video_nalu_header_size_identifies_h264_encrypted_header tests/test_stream.py::test_detect_hikvision_ps_video_nalu_header_size_probes_plausible_ciphertext_header tests/test_stream.py::test_detect_hikvision_ps_video_nalu_header_size_keeps_hevc_probe_ties -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`

The inline reply failed after the push because GitHub could not find the parent comment, but the review thread is resolved.

**Comment by RenierM26** [comment]
Addressed the latest Codex review in `445b3b5` (`Keep scanning NALs after long bodies`). The post-prefix preserved-tail guard now still skips metadata-style tail lookalikes, but allows codec-validated VCL NAL boundaries after a long body so the next NAL prefix is decrypted. Added/updated regression coverage for a preserved-tail gap before the next HEVC VCL NAL.

Validation:
- `.venv/bin/pytest tests/test_stream.py -q`
- `.venv/bin/pytest -q`
- `.venv/bin/ruff check pyezvizapi tests`
- `python -m py_compile pyezvizapi/stream.py`
- `git diff --check`
